TECH_EVAL_JIRA.md

```diff
@@ -0,0 +1,25 @@
+# Technical Evaluation Process JIRA Workflow
```

Comment:
Might we want to make it explicit that this is a _module_ technical evaluation, or is `technical evaluation` a shorthand for this?